### ISSUE TITLE
LR-tuning validation crashes when a loss schedule is active (validation batches carry epoch=None)

### ISSUE THREAD

**Issue by mcgibbon** [ai2cm/ace#1275]
## Observed

Training run [beaker 01KTG13T](https://beaker.org/ex/01KTG13T8CJHHDDFX187EGKJ6Y) (config `train-4deg-daily-v1-labels-residual-winds-anomaly-tend-reg-rollout.yaml`, launched from `80b2be8a1` on `experiment/2026-06-05-aimip-like`) crashed at epoch 45 on 2026-06-08 18:05 UTC, and crashes again identically on every resume:

```
  File "/gantry-runtime/fme/core/generics/trainer.py", line 391, in _maybe_tune_lr
    new_lr = run_lr_tuning_trial(
  File "/gantry-runtime/fme/core/generics/lr_tuning.py", line 121, in run_lr_tuning_trial
    baseline_val_loss = validate_stepper(baseline_stepper, baseline_ema)
  File "/gantry-runtime/fme/core/generics/trainer.py", line 378, in _validate_stepper
    return self._validate_stepper_callback(stepper, ema)
  File "/gantry-runtime/fme/ace/train/train.py", line 162, in validate_stepper
    run_validation_loop(
  File "/gantry-runtime/fme/core/generics/validation.py", line 61, in run_validation_loop
    stepped = stepper.train_on_batch(
  File "/gantry-runtime/fme/ace/stepper/single_module.py", line 1595, in train_on_batch
    self._loss_schedule.init_for_epoch(data.epoch)
  File "/gantry-runtime/fme/ace/stepper/loss_schedule.py", line 45, in init_for_epoch
    assert epoch is not None
AssertionError
```

Full log: last failed job [01KTM5CP](https://beaker.org/job/01KTM5CPVXFM3ZNQZ05VY9DN65) (1 failed, 4 preempted across the experiment).

## Cause

The LR-tuning trial's validation callback (`get_validate_stepper_callback` in `fme/ace/train/train.py`) never calls `set_epoch` on the validation data. When a loss schedule with milestones is active, the validation batches therefore carry `epoch=None` into `LossSchedule.init_for_epoch`, which requires an epoch. The bug triggers whenever LR tuning runs a trial while a loss schedule is configured.

`main` is still affected: `validate_stepper` at `fme/ace/train/train.py:152` still takes only `(stepper, ema)`, and `LossSchedule.init_for_epoch` on main raises `EpochNotProvidedError` ("current configuration requires epoch to be provided on BatchData during training") in the same situation.

## Fix

Fixed on `experiment/2026-06-05-aimip-like` by `1ca339ede` ("Fix epoch=None assertion in LR-tuning validation with loss schedule"): thread the current epoch through the `ValidateStepper` protocol and call `data.set_epoch(epoch)` in the callback. That commit (with its tests) needs porting to `main`.
